### ISSUE TITLE
Change package:vm_service BoundVariable.value from "dynamic" to "Response"

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#51930]
Noted at https://github.com/dart-lang/sdk/issues/51916#issuecomment-1492918215. The current type of `BoundVariable.value` is dynamic:

https://github.com/dart-lang/sdk/blob/57ba99c2d3a3688478427654f7711ed7027b163f/pkg/vm_service/lib/src/vm_service.dart#L3088-L3089

All of the types it can be implement/extend `Response` so this could be `Response`.

The problem with `dynamic` is that it's easy to pass to any function and get a runtime error (which is what #51916 was, assuming it was always `InstanceRef`).

@bkonyi notes this code is generated so might not be trivial to know the types all share a common base.

**Comment by bkonyi** [comment]
FYI @derekxu16. This isn't high priority, but we'll probably want to do a breaking release at some point to make this change. It'd probably be sufficient to change all of our instances of `dynamic` in `vm_service` classes to `Object` to avoid having to figure out the common base class of each type in a union.

**Comment by DanTup** [comment]
> It'd probably be sufficient to change all of our instances of `dynamic` in vm_service classes to `Object`

Yeah, that sounds like a much simpler change and would've also prevented the issue I hit :-)

**Comment by bkonyi** [comment]
This is a good first issue to get familiar with the `package:vm_service` code generator if anyone has some spare cycles and is interested (cc @nshahan @Markzipan @srujzs).
